/*
 * This file was automatically generated by EvoSuite
 * Thu Mar 21 01:30:12 GMT 2024
 */

package java_programs.knapsack;

import java_programs.KNAPSACK;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.Test;
import org.junit.runner.RunWith;

import static org.evosuite.runtime.EvoAssertions.verifyException;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.fail;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class KNAPSACK_ESTest extends KNAPSACK_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      int[][] intArray0 = new int[3][0];
      int[] intArray1 = new int[5];
      intArray0[0] = intArray1;
      int[] intArray2 = new int[8];
      intArray2[1] = 1;
      intArray0[1] = intArray2;
      intArray0[2] = intArray0[0];
      int int0 = KNAPSACK.knapsack(1, intArray0);
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      int[][] intArray0 = new int[1][2];
      // Undeclared exception!
      KNAPSACK.knapsack(5752, intArray0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack((-59), (int[][]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java_programs.KNAPSACK", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      int[][] intArray0 = new int[3][0];
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack((-39), intArray0);
        fail("Expecting exception: NegativeArraySizeException");
      
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java_programs.KNAPSACK", e);
      }
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      int[][] intArray0 = new int[8][7];
      int[] intArray1 = new int[4];
      intArray1[0] = (-419);
      intArray0[2] = intArray1;
      // Undeclared exception!
      try { 
        KNAPSACK.knapsack(1068, intArray0);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 1069
         //
         verifyException("java_programs.KNAPSACK", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      int[][] intArray0 = new int[2][4];
      int[] intArray1 = new int[2];
      intArray1[0] = 690;
      intArray0[0] = intArray1;
      int int0 = KNAPSACK.knapsack(690, intArray0);
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      KNAPSACK kNAPSACK0 = new KNAPSACK();
  }
}
